### PR TITLE
[SPARK-20096][Spark Submit][Minor]Expose the right queue name not null if set by --conf or configure file 

### PR DESCRIPTION
**Pull request by yaooqinn**
## What changes were proposed in this pull request?

while submit apps with -v or --verbose， we can print the right queue name, but if we set a queue name with `spark.yarn.queue` by --conf or in the spark-default.conf, we just got `null`  for the queue in Parsed arguments.
```
bin/spark-shell -v --conf spark.yarn.queue=thequeue
Using properties file: /home/hadoop/spark-2.1.0-bin-apache-hdp2.7.3/conf/spark-defaults.conf
....
Adding default property: spark.yarn.queue=default
Parsed arguments:
  master                  yarn
  deployMode              client
  ...
  queue                   null
  ....
  verbose                 true
Spark properties used, including those specified through
 --conf and those from the properties file /home/hadoop/spark-2.1.0-bin-apache-hdp2.7.3/conf/spark-defaults.conf:
  spark.yarn.queue -> thequeue
  ....
```
## How was this patch tested?

ut and local verify

### CODE REVIEW

**Comment by yaooqinn** [comment]
@srowen could please help me to verify this pr? thank you.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala
Comment (felixcheung) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: instead of `.getOrElse("default")`, i think it should be `orNull` - we don't know the default queue name of the system since it could change?

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala
Comment (felixcheung) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: the `(YARN Only)` format seems a bit unusual here? is there any other YARN only argument and how are they documented?

Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: I'm going to  rm it

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala
Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: The default value of `spark.yarn.queue` is `defualt`, I think we should follow that. Here we just print the spark submit arguments from command line and the conf file. If the queue of change by user later, they should be aware of that.

Comment (felixcheung) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: ok, I'm keep it that way for consistency, just saying that the default queue name is not necessarily "default" either.

Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: agree.

Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: would you plz trigger the test? thanks.

Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: I would be in favor of `orNull` instead of "default". `SparkSubmitArguments` reflects the arguments we set, since we don't set it, I think it would be better to be `null`. You could see other configurations here like `spark.driver.memory`, `spark.deploy.mode`, etc. Most of them indeed have default values, but here still use `null`.

Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala]: OK, orNull seems be more reasonable.

**Comment by yaooqinn** [comment]
ok to test. if i am in whitelist...

**Comment by yaooqinn** [comment]
ping @felixcheung, would you please take a look again?

**Comment by jerryshao** [comment]
The current fix LGTM, pending on committer's green light.

CC @srowen .

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala
Comment (felixcheung) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala]: use `should include ` instead?

**Review by felixcheung — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala
Comment (srowen) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala]: @yaooqinn while you're here do you want to make this last change?

Comment (yaooqinn) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala]: OK to go

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**
